### PR TITLE
Lint package YAML files

### PR DESCRIPTION
**Pull request by luhring**
- Cleans up existing YAML files, via both `yamlfmt` and manual adjustments where needed
- Adds `yamlfmt -lint` to CI checks

### CODE REVIEW

**Comment by imjasonh** [comment]
I'd like to make sure @kaniini signs off before we merge this. It kinda sucks we can't preserve empty lines, but I think being able to automate linting is more valuable than that is readable. 🤷 

I'm not sure why, but `curl.yaml` gave us trouble in https://github.com/wolfi-dev/os/pull/73 and doesn't seem to do it now.

File: envoy.yaml [hunk 1]
```diff
@@ -32,40 +31,14 @@ environment:
...
-
-
-      ./bazel/setup_clang.sh /usr
-      echo "build --config=clang" >> user.bazelrc
-
-      bazel build --verbose_failures -c opt envoy
-
-      mkdir -p ${{targets.destdir}}/usr/bin/
-      mv bazel-bin/source/exe/envoy-static ${{targets.destdir}}/usr/bin/envoy
+  - runs: "export JAVA_HOME=/usr/lib/jvm/openjdk\nmkdir -p /builder/home/.cache/bazel/_bazel_root\n\ncd envoy\n\n# these patches are only required for 1.24.0, they are already merged upstream but 1.25.0 is not yet available.\n# cannot use the melange \"patch\" pipeline as we fetch a Git repo above and can't clone into a non empty folder  \npatch -p1 < ../update_brotli.patch\npatch -p1 < ../googleurl-1.patch\npatch -p1 < ../googleurl-2.patch\npatch -p1 < ../googleurl-3.patch\npatch -p1 < ../googleurl-4.patch\npatch -p1 < ../googleurl-5.patch\npatch -p1 < ../googleurl-6.patch\n\n\n./bazel/setup_clang.sh /usr\necho \"build --config=clang\" >> user.bazelrc\n\nbazel build --verbose_failures -c opt envoy\n\nmkdir -p ${{targets.destdir}}/usr/bin/\nmv bazel-bin/source/exe/envoy-static ${{targets.destdir}}/usr/bin/envoy\n"
```
[the remark this answers is not in the document]
Comment (luhring) [reply envoy.yaml]: Great catch... I even saw that yesterday and made a note to fix it... which I then proceeded to ignore. Fixed.

**Comment by luhring** [comment]
@imjasonh 

> It kinda sucks we can't preserve empty lines

I agree, I like the empty lines.

> but I think being able to automate linting is more valuable than that is readable. 🤷

I agree here, too. If we're going to continue adding automation for these files, we'll need to be able to count on some traits of the files being consistent. If we find a way to achieve both automatability and readability, I'd love to see that 😍 

> I'm not sure why, but `curl.yaml` gave us trouble in #73 and doesn't seem to do it now.

This took me forever, but this linting exercise actually helped me find the problem with `curl`, and several others, too (like `envoy`).

The underlying cause was stray whitespace in various places. With `curl`, the issue was the trailing whitespace [here](https://github.com/wolfi-dev/os/pull/372/files#diff-927f826d42a4558885aa66ead28b5c76ea9cd2615944f60593611a12ba939ebdL48), for example.

**Comment by rawlingsj** [comment]
Yeah we have the same issues with a number of automated version update PRs.

@kaniini was suggesting we could look at adding more Styles to yaml nodes when modifying them, i.e. https://github.com/chainguard-dev/melange/blob/1425ecaae9365170199c418cc668c3efa27b8d62/pkg/renovate/bump/bump.go#L76 though I couldn't get that to work for multiline strings when I had a quick look before the break.  Maybe that's still an option.

Was chatting to @luhring yesterday, one alternative I did wonder, if there's a way we could modify yaml in place and either not rewrite the whole file or preserve existing formatting.  A quick search threw up this now archived repo https://github.com/vmware-archive/go-yaml-edit. That just seems to be a wrapper around a few standard go libraries.  Maybe it was worth a look too.  Or there maybe alternative approaches? sed? ;)

I still think yaml linting is useful, though would probably vote for ease of reading.  This repo is probably the best source of info to help folks create melange their own melange configs so I imagine people view them a lot.

Maybe we could fix the odd string concat issue for multiline strings with spaces or even look at an alternative library to help modify the yaml?

**Comment by luhring** [comment]
Yeah, this PR fixes any further weird multiline/concatenation issues, which means we'll have a better time with the Go YAML libraries now.

YAML styles are something we could consider if we don't like the way our YAML looks today. But I don't think it's related to the multiline/concatenation problem we were having.

I'm up for using more YAML libraries if they help do the job we need, but they all seem to use `gopkg.in/yaml.v3` which I tend to like and is very well supported. (It looks like `go-yaml-edit` [uses](https://github.com/vmware-archive/go-yaml-edit/blob/master/go.mod#L9) `gopkg.in/yaml.v3`, too.)

But I agree that readability is important. To be honest, I'm not 100% satisfied with `yamlfmt`. If we want to wait to lint until we have a better option, I'm okay with that, and can close this PR for now.

**Pull request closed without merging**